src/openms/source/CHEMISTRY/TheoreticalSpectrumGenerator.cpp

```diff
@@ -924,6 +924,105 @@ namespace OpenMS
     }
   }
 
+
+  void TheoreticalSpectrumGenerator::getPrefixAndSuffixIonsMZ(std::vector<float>& spectrum, const AASequence& peptide, int charge) const
+  {
+    for (Int z = charge; z >= 1; --z)
+    {
+      if (add_b_ions_)
+      {
+        addPrefixAndSuffixIons_(spectrum, peptide, Residue::BIon, z);
+      }
+      if (add_y_ions_)
+      {
+        addPrefixAndSuffixIons_(spectrum, peptide, Residue::YIon, z);
+      }
+      if (add_a_ions_)
+      {
+        addPrefixAndSuffixIons_(spectrum, peptide, Residue::AIon,  z );
+      }
+      if (add_x_ions_)
+      {
+        addPrefixAndSuffixIons_(spectrum, peptide, Residue::XIon, z);
+      }
+      if (add_c_ions_)
+      {
+        addPrefixAndSuffixIons_(spectrum, peptide, Residue::CIon, z);
+      }
+      if (add_z_ions_)
+      {
+        addPrefixAndSuffixIons_(spectrum, peptide, Residue::ZIon, z);
+      }
+    }
+
+
+    std::sort(spectrum.begin(), spectrum.end());
```

Comment:
you should be able to combine the chunks by merging. No need for sort. See https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/CHEMISTRY/TheoreticalSpectrumGenerator.cpp#L124